### DISCUSSION TITLE
Is there any way to read and write local files?

### DISCUSSION THREAD

**Discussion by pengkai** [coder/code-server#6907] [Q&A]
Is there any way to read and write local files? like using [File System API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API) to access local files.

**Comment by code-asher** [comment]
Yes, if you go to open a file you should see `Show Local` which lets you access local files on supported browsers.

**Reply by pengkai** [reply]
Thanks. It cannot be called locally via HTTP, I think it might require HTTPS.

**Reply by code-asher** [reply]
Yeah, the file system API is only allowed in secure contexts.

> Secure context: This feature is available only in [secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) (HTTPS)

Copied from https://developer.mozilla.org/en-US/docs/Web/API/File_System_API
